sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1130,6 +1130,25 @@ void ProgramManager::addImages(pi_device_binaries DeviceBinary) {
               std::make_pair(EntriesIt->name, std::move(KernelID)));
         }
       }
+      // ... and initialize associated device_global information
+      {
+        std::lock_guard<std::mutex> DeviceGlobalsGuard(m_DeviceGlobalsMutex);
+
+        auto DeviceGlobals = Img->getDeviceGlobals();
+        for (const pi_device_binary_property &DeviceGlobal : DeviceGlobals) {
+          auto Entry = m_DeviceGlobals.find(DeviceGlobal->Name);
+          assert(Entry != m_DeviceGlobals.end() &&
+                 "Device global has not been registered.");
+
+          pi::ByteArray DeviceGlobalInfo =
+              pi::DeviceBinaryProperty(DeviceGlobal).asByteArray();
+          assert(DeviceGlobalInfo.size() > 13 && "Unexpected property size");
```

Comment:
Overall LGTM.

I've not got why `DeviceGlobalInfo.size()` should be not greater than 13 (I propose this is the maximum expected size of the serialized `DeviceGlobal` property set). Could you add a comment what the magic number 13 is and why 13?